### PR TITLE
Use cookbook infrastructure

### LINKED ISSUE #335: Use Cookbook infrastructure
To simplify future maintenance and reduce dependencies, it would be good to update our GitHub Actions infrastructure for building, previewing, and link-checking to use the same actions as Pythia Cookbooks.

### PR DESCRIPTION
**Pull request by brian-rose**
Closes #335 

This updates all the GitHub Actions workflows to use the Cookbook infrastructure.

The preview may not work until this gets merged.

### CODE REVIEW

**Comment by brian-rose** [comment]
I don't understand what prettier is complaining about here.

**Comment by brian-rose** [comment]
The preview will not work until this gets merged. But I manually checked the artifact containing the html files [from the book build run here](https://github.com/ProjectPythia/pythia-foundations/actions/runs/3466812951) and they seem correct.

I guess this is ready for review!

File: .github/workflows/trigger-preview.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  find-pull-request:
+    uses: ProjectPythiaCookbooks/cookbook-actions/.github/workflows/find-pull-request.yaml@main
+  deploy-preview:
+    needs: find-pull-request
+    if: github.event.workflow_run.conclusion == 'success'
+    uses: ProjectPythiaCookbooks/cookbook-actions/.github/workflows/deploy-book.yaml@main
+    with:
+      artifact_name: book-zip-${{ needs.find-pull-request.outputs.number }}
+      destination_dir: _preview/${{ needs.find-pull-request.outputs.number }}  # deploy to subdirectory labeled with PR number
```
Comment (dcamron) [inline .github/workflows/trigger-preview.yaml:19]:
```suggestion
      destination_dir: _preview/${{ needs.find-pull-request.outputs.number }} # deploy to subdirectory labeled with PR number
```

The necessary change to satisfy prettier. I'll give this a full review later as able!

Comment (brian-rose) [reply .github/workflows/trigger-preview.yaml]: Thanks, that did the trick.

**Comment by brian-rose** [comment]
@dcamron @dcamron et al, I've been trying to get this branch updated with the latest changes in the repo and have found myself in an irreconcilable merge conflict hell.

I'm going to close this and start a complete new PR to update the Foundations build and preview infrastructure.

**Pull request closed without merging**